source/includes/api-user-access-token.rst

```diff
@@ -0,0 +1,287 @@
+           } else {
+             // An already logged in user's access token might be stale. To
+             // guarantee that the token is valid, refresh it if necessary.
+             await app.currentUser.refreshAccessToken();
+           }
+           return app.currentUser.accessToken;
+         }
+   
+   .. tab::
+      :tabid: java
+      
+      .. code-block:: java
+         :emphasize-lines: 11
+         
+         // Gets a valid user access token to authenticate requests
+         public String getValidAccessToken() {
+           // Guarantee that there's a logged in user with a valid access token
+           if (app.currentUser() == null) {
+             throw new Error(`Must be logged in to get a valid user access token`);
+           } else {
+             // An already logged in user's access token might be stale. To
+             // guarantee that the token is valid, refresh it if necessary.
+             app.currentUser().refreshAccessToken();
+           }
+           return app.currentUser().getAccessToken();
+         }
+   
+   .. tab::
+      :tabid: swift
+      
+      .. code-block:: swift
+         :emphasize-lines: 14
+         
+         enum AccessTokenError: Error {
+           case notLoggedIn(String)
+         }
+         
+         // Gets a valid user access token to authenticate requests
+         func getValidAccessToken() throws -> String {
+           // Guarantee that there's a logged in user with a valid access token
+           guard let user = app.currentUser else {
+             throw AccessTokenError.notLoggedIn("Must be logged in to get a valid user access token")
+           }
+           // An already logged in user's access token might be stale. To
+           // guarantee that the token is valid, refresh it if necessary.
+           user.refreshCustomData()
+           return user.accessToken!
+         }
+   
+   .. tab::
+      :tabid: c-sharp
+      
+      .. code-block:: csharp
+         :emphasize-lines: 15
+         
+         // Gets a valid user access token to authenticate requests
+         public async Task<string> GetValidAccessToken()
+         {
+             // Guarantee that there's a logged in user with a valid access token
+             if (app.CurrentUser == null)
+             {
+                 throw new Exception("Must be logged in to get a valid user access token");
+             }
+             else
+             {
+                 // An already logged in user's access token might be stale. To
+                 // guarantee that the token is valid, refresh it if necessary.
+                 await app.CurrentUser.RefreshCustomDataAsync();
+             }
+             return app.CurrentUser.AccessToken;
+         }
+
+Get a User Access Token from the Client API
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can get an access token by sending an authentication request
+directly to the App Services Client API. The authentication endpoint
```

Comment:
'App Services Client API' is a new one to me. should we have docs on this?

Reply:
also why are we documenting the Client API here when we don't anywhere else in the App Services docs?

Reply:
do we need an OpenAPI spec for this?

Reply:
Product decision to leave it undocumented - we have one endpoint here because it's how you authenticate via HTTP

Reply:
Actually we also use the client API for GraphQL. Let's bring it up with product next sync meeting.

Reply:
Changed to "Over HTTP" for now

Reply:
> Product decision to leave it undocumented - we have one endpoint here because it's how you authenticate via HTTP

if product wants, we could add a page in the [Authentication & Users section](https://www.mongodb.com/docs/atlas/app-services/authentication/) titled something like 'Authenticate over HTTP'. this page could *maybe* just be an OpenAPI spec w some additional info at the top.

Reply:
Yeah this does seem reasonable